CMakeLists.txt

```diff
@@ -121,21 +123,38 @@ add_executable (
   terminal/UserTerminalHandler.cpp
   terminal/UserTerminalRouter.cpp
   )
-target_link_libraries (
-  etserver
-  LINK_PUBLIC
-  TerminalCommon
-  EternalTCP-static
-  ${CMAKE_THREAD_LIBS_INIT}
-  ${PROTOBUF_LIBRARIES}
-  ${GLOG_LIBRARIES}
-  ${GFLAGS_LIBRARIES}
-  ${sodium_LIBRARY_RELEASE}
-  ${SELINUX_LIBRARIES}
-  ${UTEMPTER_LIBRARIES}
-  resolv
-  util
+IF(!FREEBSD)
```

Comment:
Instead of putting the logic here, can you create a new cmake variable called CORE_LIBRARIES and set it to either resolv + util or just util depending on IF(FREEBSD)?